main.tf

```diff
@@ -4,226 +4,81 @@ provider "aws" {
   region     = "${var.region}"
 }
 
-module "elasticsearch" {
-  source             = "github.com/everydayhero/terraform-elasticsearch"
-  name               = "logs"
-  access_key         = "${var.access_key}"
-  secret_key         = "${var.secret_key}"
-  region             = "${var.region}"
-  key_name           = "${var.key_name}"
-  subnet_ids         = "${var.subnet_ids}"
-  vpc_id             = "${var.vpc_id}"
-  vpc_cidr           = "${var.vpc_cidr}"
-  instance_type      = "${var.instance_type}"
-  image_id           = "${var.image_id}"
-  cluster_size       = "${var.cluster_size}"
-  volume_size_data   = "${var.volume_size}"
-  ssh_keys           = "${var.ssh_keys}"
-  replicas           = "1"
-}
+data "aws_ami" "es" {
+  most_recent = true
+  executable_users = ["all", "self"]
 
-resource "aws_iam_role" "function" {
-  name = "LogsFunction"
-  assume_role_policy = <<POLICY
-{
-  "Version": "2012-10-17",
-  "Statement": [
-    {
-      "Sid": "",
-      "Effect": "Allow",
-      "Principal": {
-        "Service": "lambda.amazonaws.com"
-      },
-      "Action": "sts:AssumeRole"
-    }
-  ]
-}
-POLICY
-}
-
-resource "aws_iam_role_policy" "function" {
-  name = "LogsFunctionExecution"
-  role = "${aws_iam_role.function.id}"
-  policy = <<POLICY
-{
-  "Version": "2012-10-17",
-  "Statement": [
-    {
-      "Effect": "Allow",
-      "Action": [
-        "logs:CreateLogGroup",
-        "logs:CreateLogStream",
-        "logs:PutLogEvents",
-        "ec2:CreateNetworkInterface",
-        "ec2:DescribeNetworkInterfaces",
-        "ec2:DeleteNetworkInterface",
-        "kinesis:GetRecords",
-        "kinesis:GetShardIterator",
-        "kinesis:DescribeStream",
-        "kinesis:ListStreams"
-      ],
-      "Resource": "*"
-    }
-  ]
-}
-POLICY
-}
-
-resource "aws_security_group" "function" {
-  name = "LogsFunction"
-  description = "Allows function access to elasticsearch"
-
-  vpc_id = "${var.vpc_id}"
-
-  egress {
-    from_port = 80
-    to_port   = 80
-    protocol  = "tcp"
-    security_groups = ["${module.elasticsearch.security_group_id}"]
+  filter {
+    name = "name"
+    values = ["*/hvm-ssd/ubuntu-xenial-16.04-amd64-server*"]
```

Comment:
No it shouldn't. It's a generic AMI finder while we specify the actual AMI we want to use during build time.